grakn-graph/src/main/java/ai/grakn/graph/internal/RelationTypeImpl.java

```diff
@@ -95,14 +98,34 @@ public RelationType deleteHasRole(RoleType roleType) {
 
         RoleTypeImpl roleTypeImpl = (RoleTypeImpl) roleType;
         //Add castings of roleType to make sure relations are still valid
-        roleTypeImpl.castings().forEach(casting -> getGraknGraph().getConceptLog().putConcept(casting));
+        roleTypeImpl.castings().forEach(casting -> getGraknGraph().getConceptLog().trackConceptForValidation(casting));
 
         //Add the Role Type itself
-        getGraknGraph().getConceptLog().putConcept(roleTypeImpl);
+        getGraknGraph().getConceptLog().trackConceptForValidation(roleTypeImpl);
 
         //Add the Relation Type
-        getGraknGraph().getConceptLog().putConcept(roleTypeImpl);
+        getGraknGraph().getConceptLog().trackConceptForValidation(roleTypeImpl);
+
+        //Remove from internal cache
+        if(cachedHasRoles.isPresent()) cachedHasRoles.get().remove(roleType);
```

Comment:
Perhaps we should have a helper method for this on the cache. It will also make sure you don't forget the `ifPresent` check.